File: DivideFraction.py
from MultiplyFraction import TimesFraction

# A class with a parent
class DivFraction(TimesFraction):

    # A constructor
    def __init__(self, fractions):
        super().__init__(fractions)


    # Returns a dictionary of the new fraction
    def doDivide(self):
        # Invert every other fraction but the first
        for count in range(1, len(self.numerators)):
            self.temp = self.numerators[count]
            self.numerators[count] = self.denominators[count]
            self.denominators[count] = self.temp


        return self.doMultiply()

File: DivideFraction.vb
Public Class DivideFraction
    Inherits MultiplyFraction

    ' Returns a dictionary Of the New fraction
    Public Function doDivide() As Dictionary(Of String, Integer)
        ' Invert every other fraction but the first
        For index = 1 To numerators.Count - 1
            Dim temp = numerators(index)
            numerators(index) = denominators(index)
            denominators(index) = temp
        Next index

        Return doMultiply()
            End Function

End Class
